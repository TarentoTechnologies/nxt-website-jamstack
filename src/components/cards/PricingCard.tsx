import checkIcon from "../../../static/images/green-check.svg";
import { bgWhite } from "../../styles/style-guide/ColorGuide.module.css";
import { cardHeader } from "../../styles/style-guide/Typography.module.css";
import {
  basePriceBox,
  optional,
  pricingBox,
  supportBar,
  tickStyle,
} from "./PricingCard.module.css";

interface PricingCardProps {
  title: string;
  logo: any;
  description: string;
  basePrice?: string;
  supportPrice?: string;
  supportText?: string;
  supportSubText?: string;
  baseList?: string[];
  supportList?: string[];
}

export const PricingCard = ({
  title,
  logo,
  description,
  baseList,
  basePrice,
  supportList,
  supportPrice,
  supportSubText,
  supportText,
}: PricingCardProps) => {
  return (
    <div className={`col-md-4 ${bgWhite} mb-4`}>
      <div className={`${pricingBox} p-4`}>
        <div className={` d-flex align-items-center ps-3`}>
          <img src={logo} alt={title} />
          <div className={`${cardHeader} ms-4`}>{title}</div>
        </div>
        <div className="mt-5 mb-5 ps-3">{description} </div>
        <div className={basePriceBox}>{basePrice}</div>
        {baseList?.map((item, index) => (
          <div className="d-flex mt-3 mb-4 ps-3">
            <div className="me-3">
              <img src={checkIcon} className={tickStyle} />
            </div>
            <span>{item}</span>
          </div>
        ))}

        <div className={supportBar}>
          <p>{supportPrice} </p>
          <span className="floatRight">
            {" "}
            {supportText}
            <span className={optional}>{supportSubText}</span>
          </span>
        </div>
        {supportList?.map((item, index) => (
          <div className="d-flex mt-3 mb-4 ps-3">
            <div className="me-3">
              <img src={checkIcon} className={tickStyle} />
            </div>
            <span>{item}</span>
          </div>
        ))}
      </div>
    </div>
  );
};
