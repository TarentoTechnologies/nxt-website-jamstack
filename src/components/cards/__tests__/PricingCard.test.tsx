import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PricingCard } from "../PricingCard";

const baseListData = ["Full perpetual license", "100 hours for implementation"];

describe("Test if the PricingCard component is working as expected", () => {
  it("<PricingCard /> matches snapshot", () => {
    const component = render(
      <PricingCard
        title={"RAIN"}
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/rain_with_circle_ae2744ceec.svg"
        }
        description={
          "Real time analytics and insights from your business data."
        }
        spTitle={""}
        baseList={baseListData}
        basePrice="€25K"
        supportList={baseListData}
        supportPrice="€3000 per year"
        supportSubText="(Optional)"
        supportText="Support"
        withBtn={false}
        CTAlabel={"Contact us"}
        CTAlink={"/"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PricingCard /> renders without exploding", () => {
    const component = render(
      <PricingCard
        title={"RAIN"}
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/rain_with_circle_ae2744ceec.svg"
        }
        description={
          "Real time analytics and insights from your business data."
        }
        spTitle={""}
        baseList={baseListData}
        basePrice="€25K"
        supportList={baseListData}
        supportPrice="€3000 per year"
        supportSubText="(Optional)"
        supportText="Support"
        withBtn={false}
        CTAlabel={"Contact us"}
        CTAlink={"/"}
      />
    );
    expect(component.getByTestId("pricing-card")).toBeInTheDocument();
  });
});
