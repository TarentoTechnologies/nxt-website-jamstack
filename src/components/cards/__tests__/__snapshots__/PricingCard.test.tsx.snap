// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PricingCard component is working as expected <PricingCard /> matches snapshot 1`] = `
<div>
  <div
    class="col-sm-12 col-md-6 col-lg-5 col-xl-4"
    data-testid="pricing-card"
  >
    <div
      class="pricingBox bgWhite p-4"
    >
      <div
        class="levelOneBox"
      >
        <div
          class="d-flex align-items-center ps-3 "
        >
          <img
            alt="RAIN"
            data-aos="fade-up"
            src="https://strapi-stage-nxt.tarento.com/uploads/rain_with_circle_ae2744ceec.svg"
          />
          <div
            class="cardHeader ms-4"
            data-aos="fade-up"
          >
            RAIN
          </div>
        </div>
        <div
          class="mt-4 mb-3 ps-3"
          data-aos="fade-up"
        >
          Real time analytics and insights from your business data.
        </div>
      </div>
      <div
        class="basePriceBox"
        data-aos="fade-up"
      >
        €25K
      </div>
      <div
        class="d-flex mt-3 mb-4 ps-3"
      >
        <div
          class="me-3"
        >
          <img
            class="tickStyle"
            data-aos="fade-up"
            src="test-file-stub"
          />
        </div>
        <span
          data-aos="fade-up"
        />
      </div>
      <div
        class="d-flex mt-3 mb-4 ps-3"
      >
        <div
          class="me-3"
        >
          <img
            class="tickStyle"
            data-aos="fade-up"
            src="test-file-stub"
          />
        </div>
        <span
          data-aos="fade-up"
        />
      </div>
      <div
        class="supportBar"
      >
        <p
          data-aos="fade-up"
        >
          €3000 per year
           
        </p>
        <span
          class="floatRight"
          data-aos="fade-up"
        >
           
          Support
          <span
            class="optional"
            data-aos="fade-up"
          >
            (Optional)
          </span>
        </span>
      </div>
      <div
        class="d-flex mt-3 mb-4 ps-3"
      >
        <div
          class="me-3"
        >
          <img
            class="tickStyle"
            data-aos="fade-up"
            src="test-file-stub"
          />
        </div>
        <span
          data-aos="fade-up"
        />
      </div>
      <div
        class="d-flex mt-3 mb-4 ps-3"
      >
        <div
          class="me-3"
        >
          <img
            class="tickStyle"
            data-aos="fade-up"
            src="test-file-stub"
          />
        </div>
        <span
          data-aos="fade-up"
        />
      </div>
    </div>
  </div>
</div>
`;
